integration/integration_test.go

```diff
@@ -7223,6 +7226,160 @@ func testSFTP(t *testing.T, suite *integrationTestSuite) {
 	require.Equal(t, testFilePath, sftpEvent.GetString(events.SFTPPath))
 }
 
+func testAgentlessConnection(t *testing.T, suite *integrationTestSuite) {
+	// create Teleport instance
+	teleInst := suite.newTeleport(t, nil, true)
+	t.Cleanup(func() {
+		teleInst.StopAll()
+	})
+
+	tc, err := teleInst.NewClient(helpers.ClientConfig{
+		Login:   suite.Me.Username,
+		Cluster: helpers.Site,
+		Host:    Host,
+	})
+	require.NoError(t, err)
+
+	// get OpenSSH CA public key and create host certs
+	ctx := context.Background()
+	authClient := teleInst.Process.GetAuthServer()
+	openSSHCA, err := authClient.GetCertAuthority(ctx, types.CertAuthID{
+		Type:       types.OpenSSHCA,
+		DomainName: helpers.Site,
+	}, false)
+	require.NoError(t, err)
+
+	caCheckers, err := sshutils.GetCheckers(openSSHCA)
+	require.NoError(t, err)
+
+	key, err := native.GeneratePrivateKey()
+	require.NoError(t, err)
+
+	const nodeName string = "agentless-node"
+	hostCertBytes, err := authClient.GenerateHostCert(
+		ctx,
+		key.MarshalSSHPublicKey(),
+		"",
+		"",
+		[]string{nodeName, Loopback},
+		helpers.Site,
+		types.RoleNode,
+		0,
+	)
+	require.NoError(t, err)
+
+	hostCert, err := apisshutils.ParseCertificate(hostCertBytes)
+	require.NoError(t, err)
+	signer, err := ssh.NewSignerFromSigner(key.Signer)
+	require.NoError(t, err)
+	hostKeySigner, err := ssh.NewCertSigner(hostCert, signer)
+	require.NoError(t, err)
+
+	// start SSH server
+	sshAddr := startSSHServer(t, caCheckers, hostKeySigner)
+
+	// create node resource
+	_, err = authClient.UpsertNode(ctx, &types.ServerV2{
+		Kind:    types.KindNode,
+		SubKind: types.SubKindOpenSSHNode,
+		Version: types.V2,
+		Metadata: types.Metadata{
+			Name: nodeName,
+		},
+		Spec: types.ServerSpecV2{
+			Addr:     sshAddr,
+			Hostname: nodeName,
+		},
+	})
+	require.NoError(t, err)
+
+	// wait for node resource to be written to the backend
+	timedCtx, cancel := context.WithTimeout(ctx, time.Second)
+	t.Cleanup(cancel)
+	w, err := authClient.NewWatcher(timedCtx, types.Watch{
+		Name: "node-create watcher",
+		Kinds: []types.WatchKind{
+			{
+				Kind: types.KindNode,
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	var nodeCreated bool
+	for !nodeCreated {
```

Comment:
```suggestion
	for nodeCreated := false; !nodeCreated; {
```

The smaller the scope, the better.

Reply:
Good point, done